### DISCUSSION TITLE
monitor souce folder with depth

### DISCUSSION THREAD

**Discussion by mewu3** [axkibe/lsyncd#723] [Problems]
it's possible to monitor only the first layer of my source folder? I trying to monitor the source and react on creation of different files (base on extension, etc.) and folder. 
Specifically for the folder, i only want to monitor the first layer of my source, I do not want to monitor over the deeper depth. 
I search a bit online but couldn't find any one have ever thought of this ...

**Comment by ferdnyc** [comment]
If you want to monitor only items added directly to `/some/srcdir/`, I **think** you can just use a non-recursive sync:

```js
sync {
    default.rsync[ssh],
    source="/some/srcdir",
    recursive=false,
}
```

as in the [Auto-Image-Magic example](https://lsyncd.github.io/lsyncd/manual/examples/auto-image-magic/) that automatically converts files added to `magicdir`.

But if you need to monitor `/some/srcdir/<anydirectory>/*` non-recursively, that's probably trickier.

You could _definitely_ do it with the proper layer-1 scripting, again like in the Auto-Image-Magic example.

**Reply by ferdnyc** [reply]
_Generally_, though, it's usually just easier to sync an entire tree, and _not_ store anything inside the tree on one end that you don't want on the other end.

IOW, if you're going to sync `$HOME/mysyncdir/` and anything directly inside it, but only one level deep, then why does `$HOME/mysyncdir/somesubdir/adeepersubdir/` have to be stored inside `$HOME/mysyncdir/somesubdir/`? That seems confusing.

I'm sure there are situations where it might make sense to do that, and I accept that yours is one of those situations. But I bet they're rarer than you might think. If `adeepersubdir` doesn't need to be synced, why store it inside `$HOME/mysyncdir/` at all?

**Comment by mewu3** [comment]
thank you a lot for your answer !
